include/GafferImage/ImageReader.h

```diff
@@ -110,10 +110,18 @@ class ImageReader : public ImageNode
 		Gaffer::StringPlug *colorSpacePlug();
 		const Gaffer::StringPlug *colorSpacePlug() const;
 
+		Gaffer::StringPlug *colorSpaceMetadataNamePlug();
+		const Gaffer::StringPlug *colorSpaceMetadataNamePlug() const;
+
 		virtual void affects( const Gaffer::Plug *input, AffectedPlugsContainer &outputs ) const;
 
 		static size_t supportedExtensions( std::vector<std::string> &extensions );
 
+		typedef boost::function<const std::string ( const IECore::CompoundData * )> DefaultColorSpaceFunction;	
+		static void registerDefaultColorSpace( DefaultColorSpaceFunction f );
```

Comment:
Are you 100% sure about that? My suspicion is that the `format` data member will be float, but that there is also an "OIIO:BitsPerSample" metadata entry which is triggering the "uint" code path.